lldb/source/Symbol/LocateSymbolFile.cpp

```diff
@@ -396,8 +398,22 @@ Symbols::LocateExecutableSymbolFile(const ModuleSpec &module_spec,
       }
     }
   }
-
-  return LocateExecutableSymbolFileDsym(module_spec);
+  FileSpec dsym_bundle = LocateExecutableSymbolFileDsym(module_spec);
+  if (dsym_bundle)
+    return dsym_bundle;
+
+  // If we didn't find anything by looking locally, let's try Debuginfod.
+  if (module_uuid.IsValid() && llvm::canUseDebuginfod()) {
+    llvm::object::BuildID build_id(module_uuid.GetBytes());
+    llvm::Expected<std::string> result =
+        llvm::getCachedOrDownloadDebuginfo(build_id);
+    if (result)
+      return FileSpec(*result);
+    // An error is just fine, here...
+    consumeError(result.takeError());
```

Comment:
log this to some log channel instead of consuming it

Reply:
That was my reason for just consuming it in the first place. I think adding a "debuginfod_failure_log" setting makes sense here. And I just signed myself up for more work :/

Reply:
Looks like gdb has a `set debuginfod verbose on/off` setting for this. There's also the generic `DEBUGINFOD_VERBOSE` environment variable. If set, this triggers remote logging to stderr.

The environment variable is something we generally want to support for debuginfod lookups; that could probably safely live in the debuginfod client library itself, as it does with libdebuginfod. If LLDB had its own setting for this, it may be able to invoke a setter in the client library, but current everything in the client library is global, so there may be some threading gotchas there.

For reference, here's the gdb settings for debuginfod: https://sourceware.org/gdb/current/onlinedocs/gdb.html/Debuginfod-Settings.html
And the environment variables:
https://manpages.debian.org/testing/debuginfod/debuginfod.8.en.html#ENVIRONMENT_VARIABLES

Reply:
@mysterymath that environment variable is used by the server, not the client.

In the interest of moving forward, I'd propose that this code is left as-is, and I'll add logging in libdebuginfod (separate from LLDB, and honestly, probably more useful) in a subsequent PR. I'll probably need this capability for my next steps, anyway (source code is next on my list)

Reply:
> @mysterymath that environment variable is used by the server, not the client.
Ah, I should have linked to the debuginfod-find man page instead; pardon my laziness. It is used by the client, it's just that the server also includes a client for federation purposes.

See https://sourceware.org/git/?p=elfutils.git;a=blob;f=debuginfod/debuginfod-client.c;h=dcf0503163fd73e126a2197d278f363fd6839810;hb=HEAD#l1881

Reply:
I'm going to add better diagnostics to the llvm library instead, thus I didn't make a change to this in the updated diff...